File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/49862-SharePoint 2010 101 Code Samples/SharePoint 2010 Creating Custom SharePoint Service Applications/VB/WCF_CustomServiceApplication/DayNamerServiceProxy.vb
Imports Microsoft.SharePoint.Administration

Namespace Client

    'This class runs on the Web-Front End servers and 
    'connects to the WCF service on the application servers
    'This is why web parts etc. don't need to know where
    'the service application actually runs. Make sure that
    'the SupportedServiceApplication GUID matches the one
    'in the DayNamerServiceApplication class.
    <System.Runtime.InteropServices.Guid("5FB8E4AA-9C30-43B0-93C3-CEFEA5A18110")> _
    <SupportedServiceApplication( _
        "3CB59033-C70F-4699-BCA9-443DE4FF0D97", _
        "1.0.0.0", _
        GetType(DayNamerServiceApplicationProxy))> _
    Public Class DayNamerServiceProxy
        Inherits SPIisWebServiceProxy
        Implements IServiceProxyAdministration

#Region "Constructors"

        Public Sub New()
            MyBase.New()
        End Sub

        Public Sub New(ByVal farm As SPFarm)
            MyBase.New(farm)
        End Sub

#End Region

        Public Function CreateProxy(ByVal serviceApplicationProxyType As Type, _
                                    ByVal name As String, _
                                    ByVal serviceApplicationUri As Uri, _
                                    ByVal provisioningContext As SPServiceProvisioningContext) _
                                As SPServiceApplicationProxy _
                                Implements IServiceProxyAdministration.CreateProxy

            If Not serviceApplicationProxyType Is GetType(DayNamerServiceApplicationProxy) Then
                Throw New NotSupportedException()
            End If

            Return New DayNamerServiceApplicationProxy(name, Me, serviceApplicationUri)

        End Function

        Public Function GetProxyTypeDescription(ByVal serviceApplicationProxyType As Type) As SPPersistedTypeDescription _
            Implements IServiceProxyAdministration.GetProxyTypeDescription

            Return New SPPersistedTypeDescription("Day Namer Service Proxy", "Custom service application proxy providing simple day names.")

        End Function

        Public Function GetProxyTypes() As Type() Implements IServiceProxyAdministration.GetProxyTypes
            Dim returnType As Type() = {GetType(DayNamerServiceApplicationProxy)}
            Return returnType
        End Function

        Public Overrides Function GetCreateProxyLink(ByVal serviceApplicationProxyType As System.Type) As SPAdministrationLink _
            Implements IServiceProxyAdministration.GetCreateProxyLink

            Return MyBase.GetCreateProxyLink(serviceApplicationProxyType)
        End Function

    End Class

End Namespace

File: Microsoft Office Developer Documentation Team/SharePoint 2010 101 Code Samples/49862-SharePoint 2010 101 Code Samples/SharePoint 2010 Creating Custom SharePoint Service Applications/C#/WCF_CustomServiceApplication/DayNamerServiceProxy.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.SharePoint.Administration;

namespace WCF_CustomServiceApplication.Client
{
    //This class runs on the Web-Front End servers and 
    //connects to the WCF service on the application servers
    //This is why web parts etc. don't need to know where
    //the service application actually runs. Make sure that
    //the SupportedServiceApplication GUID matches the one
    //in the DayNamerServiceApplication class.
    [System.Runtime.InteropServices.Guid("841D1124-83CD-483A-9043-09B7A1CB7505")]
    [SupportedServiceApplication("4D45F3BB-5996-4EF1-855E-BD68D918A3A6", 
                                "1.0.0.0", 
                                typeof(DayNamerServiceApplicationProxy))]
    public class DayNamerServiceProxy: SPIisWebServiceProxy, IServiceProxyAdministration
    {
        #region Constructors

        public DayNamerServiceProxy()
            : base() { }
        public DayNamerServiceProxy(SPFarm farm)
            : base(farm) { }

        #endregion

        public SPServiceApplicationProxy CreateProxy(Type serviceApplicationProxyType, string name, Uri serviceApplicationUri, SPServiceProvisioningContext provisioningContext)
        {
            if (serviceApplicationProxyType != typeof(DayNamerServiceApplicationProxy))
                throw new NotSupportedException();

            return new DayNamerServiceApplicationProxy(name, this, serviceApplicationUri);
        }

        public SPPersistedTypeDescription GetProxyTypeDescription(Type serviceApplicationProxyType)
        {
            return new SPPersistedTypeDescription("Day Namer Service Proxy", "Custom service application proxy providing simple day names.");
        }

        public Type[] GetProxyTypes()
        {
            return new Type[] { typeof(DayNamerServiceApplicationProxy) };
        }
    }
}
